clifford/test/test_trig_functions.py

```diff
@@ -0,0 +1,86 @@
+
+from .. import Cl
+import numpy as np
+import pytest
+
+
+class TestScalarProperties:
+    @pytest.fixture()
+    def element(self):
+        alg, blades = Cl(0, 0, 0)
+        return alg.scalar
+
+    @pytest.mark.parametrize('np_func', [np.sin,
+                                         np.cos,
+                                         np.tan,
+                                         np.sinh,
+                                         np.cosh,
+                                         np.tanh])
+    def test_trig(self, element, np_func):
+        for x in np.linspace(0, 2*np.pi, 100):
+            assert abs(np_func(x*element).value[0] - np_func(x)) < 1E-10
+
+
+class TestDualNumberProperties:
+    @pytest.fixture()
+    def element(self):
+        alg, blades = Cl(0, 0, 1)
+        return alg.scalar, blades['e1']
+
+    @pytest.mark.parametrize('func, deriv_func', [(np.sin, np.cos),
+                                                  (np.cos, lambda x: -np.sin(x)),
+                                                  (np.tan, lambda x: (1/np.cos(x)**2)),
+                                                  (np.sinh, np.cosh),
+                                                  (np.cosh, np.sinh),
+                                                  (np.tanh, lambda x: (1 - np.tanh(x)**2))])
+    def test_derivatives(self, element, func, deriv_func):
+        for x in np.linspace(0, 2*np.pi, 10):
```

Comment:
How do these fare outside `[0, 2*pi]`? I'd be inclined to expand this to at least include some negative values,
```suggestion
        for x in np.linspace(-2*np.pi, 2*np.pi, 13):
```